cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/image/AzureImageSetupService.java

```diff
@@ -96,8 +97,15 @@ public ImageStatusResult checkImageStatus(AuthenticatedContext ac, CloudStack st
         }
     }
 
-    private Optional<VirtualMachineCustomImage> findVirtualMachineCustomImage(Image image, String imageResourceGroupName, AzureClient client) {
-        return azureManagedImageService.findVirtualMachineCustomImage(imageResourceGroupName, image.getImageName(), client);
+    private Optional<VirtualMachineCustomImage> findVirtualMachineCustomImage(Image image, String imageResourceGroupName, AzureClient client,
+            AuthenticatedContext ac) {
+        String imageName = getImageName(image.getImageName(), ac.getCloudContext().getLocation().getRegion().getRegionName());
+        return azureManagedImageService.findVirtualMachineCustomImage(imageResourceGroupName, imageName, client);
+    }
+
+    private String getImageName(String fromVhdUri, String region) {
```

Comment:
This piece is present already in class AzureImageService. It might make sense to extract them.